### ISSUE TITLE
Traning time with ResNet50 networks

### ISSUE THREAD

**Issue by lianqing01** [tingxueronghua/pytorch-classification-advprop#1]
Dear author, 

Thanks for your kindly implement advprop. I am wondering that can you provide use the training time of AdvProp with ResNet-50 architecture. 

Best regards

**Comment by tingxueronghua** [comment]
Of course, feel free to ask any questions. Under the experiment settings mentioned in the readme, training ResNet-50 using AdvProp costs 50-60 minutes per epoch. In comparison, original training (without AdvProp) costs around 20 minutes per epoch.

**Comment by lucasliunju** [comment]
Hi, thanks for your contribution. I would like to ask the value of final training loss (the batch data includes clean data and adv data). In addition, I want to ask the main reason increasing training time for advprop + resnet. In my opinion, it's the batch data become larger (2 time). How about the calculation of gradient in attack model.

Thanks for your help.

**Comment by tingxueronghua** [comment]
Thanks for your attention.
Around 0.99 under the experiment settings mentioned in readme. 
The reason you mentioned is one reason. Another reason is about the PGD attack. When we do PGD attack with epsilon as 1, we have to do one more forward and backward propagation to get the gradients on the input images. Then you can see it should be at least three times.
